content/automate/ManageIQ/Infrastructure/VM/Transform/StateMachines/VmImportWaitForVm.class/__methods__/wait_for_imported_vm.rb

```diff
@@ -1,36 +1,42 @@
+#
+# Description: This method checks if the VM has been successfully imported
+#
+
 module ManageIQ
   module Automate
-    module System
-      module Event
-        module EmsEvent
-          module RHEVM
-            class UpdateVmImportStatus
```

Comment:
@pkliczewski @smelamud -  @mkanoor, @agrare and I discussed various aspects of the issue.
I have a few suggestions:
1. Change the event refresh call.
The RHEVM event IMPORTEXPORT_IMPORT_VM instance currently calls for a refresh with the following: 
/System/event_handlers/event_action_refresh?target=src_vm

It should call this instead:
/System/event_handlers/event_action_refresh_new_target?target=src_vm

2. This new state machine is not necessary. The original import state machine will be pick up the VM on a retry and the failure will be caught by the state machine when it exceeds the specified number of retries.

3. Remove the custom_get logic from the original state machine wait_for_vm_import method.
    status = vm.custom_get(:import_status)
The method is a success if there is a VM, otherwise setup the retry.